tests/azure_iot_ut.c

```diff
@@ -0,0 +1,172 @@
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );  
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesAppend_Failure( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucTestKey ) ),
+                      eAzureIoTSuccess );
+
+    /* Failed for NULL key passed */
+    assert_int_not_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                            NULL, 0, ucTestValue,
+                                                            sizeof( ucTestValue ) - 1 ),
+                          eAzureIoTSuccess ); 
+
+    /* Failed for NULL value passed */
+    assert_int_not_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                            NULL, 0, ucTestValue,
+                                                            sizeof( ucTestValue ) - 1 ),
+                          eAzureIoTSuccess ); 
+
+    /* Failed for bigger data append */
+    assert_int_not_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                            ucTestKey, sizeof( ucTestKey ) - 1,
+                                                            ucTestValue, sizeof( ucTestValue ) - 1 ),
+                          eAzureIoTSuccess );      
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesAppend_Success( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                        ucTestKey, sizeof( ucTestKey ) - 1,
+                                                        ucTestValue, sizeof( ucTestValue ) - 1 ),
+                      eAzureIoTSuccess );   
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesFind_Failure( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+    const uint8_t * pucOutValue;
+    uint32_t ulOutValueLength;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
```

Comment:
Same feedback here & line below about having the /* ... */ explaining test, even if obvious.  Comments any other places this may be missing.